### PR TITLE
feat: load puppeteer faster, run stores in parallel

### PR DESCRIPTION
**Pull request by arickho**
### Description

- Each store scraping now re-triggers itself oppose to waiting for all stores to process 
- Ability to switch between HEADLESS on/off
- Convert env var booleans to correct data type
- Use StealthPlugin + AdblockerPlugin (using pupeteer-extra lib)
- Only initialize Puppeteer once

### Testing

N/A

### New dependencies

```
    "async": "^3.2.0",
    "puppeteer-extra": "^3.1.15",
    "puppeteer-extra-plugin-adblocker": "^2.11.6",
    "puppeteer-extra-plugin-stealth": "^2.6.1",
```

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,29 +1,57 @@
...
+				await lookup(browser, store);
+			} catch (error) {
+				// Ignoring errors; more than likely due to rate limits
+				Logger.error(error);
+			} finally {
+				cb();
+				q.push(store);
+			}
+		}, Config.rateLimitTimeout);
+	}, Stores.length);
```
Comment (arickho) [inline src/index.ts:39]: Limits queue concurrency to the number of Stores configured

**Comment by arickho** [comment]
@jef this should fix:

https://github.com/jef/nvidia-snatcher/issues/61
https://github.com/jef/nvidia-snatcher/issues/68
https://github.com/jef/nvidia-snatcher/issues/57

**Comment by jef** [comment]
I'll review when I get back home. Thanks! What's your discord?

**Comment by arickho** [comment]
> I'll review when I get back home. Thanks! What's your discord?

Arview#6463

**Comment by arickho** [comment]
Merge conflicts fixed after v1.4.1 release.

File: src/config.ts [hunk 2]
```diff
@@ -34,27 +34,36 @@ const notifications = {
 		accessToken: process.env.TELEGRAM_ACCESS_TOKEN ?? '',
 		chatId: process.env.TELEGRAM_CHAT_ID ?? ''
 	},
-	test: process.env.NOTIFICATION_TEST ?? 'false'
+	test: process.env.NOTIFICATION_TEST === 'true'
```
Comment (jef) [inline src/config.ts:37]: I really like this.

File: src/config.ts [hunk 3]
```diff
@@ -34,27 +34,36 @@ const notifications = {
...
 };
 
-const openBrowser = process.env.OPEN_BROWSER ?? 'true';
 const rateLimitTimeout = Number(process.env.RATE_LIMIT_TIMEOUT) ?? 5000;
+
 const stores = process.env.STORES ? process.env.STORES.split(',') : ['nvidia'];
+
+const openBrowser = process.env.OPEN_BROWSER === 'true';
+
+const isHeadless = process.env.HEADLESS === 'true';
```
Comment (jef) [inline src/config.ts:54]: Maybe I'll refactor this after this gets merged in to put into a new object `Config.browser` now that this configuration is growing quite a bit.

File: src/index.ts [hunk 4]
```diff
@@ -1,33 +1,57 @@
...
-import {Stores} from './store/model';
+import {Store, Stores} from './store/model';
 import {Logger} from './logger';
 import {sendNotification} from './notification';
 import {lookup} from './store';
-import puppeteer from 'puppeteer';
+import async from 'async';
+
+puppeteer.use(stealthPlugin());
+puppeteer.use(adblockerPlugin({blockTrackers: true}));
```
Comment (jef) [inline src/index.ts:12]: Nice

**Review by jef — APPROVED** [review]
I love the changes. Thanks for great improvements here.

**Pull request merged**